govcd/openapi_endpoints.go

```diff
@@ -12,7 +12,9 @@ import (
 
 // endpointMinApiVersions holds mapping of OpenAPI endpoints and API versions they were introduced in.
 var endpointMinApiVersions = map[string]string{
-	types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointRoles: "31.0",
+	types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointRoles:              "31.0",
+	types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcComputePolicies: "32.0",
+	types.OpenApiEndpointAssignedComputePolicies:                            "33.0",
```

Comment:
well, this "pattern" isn't right then, first use case and link is different (id inside xx/id/xx) and can't follow provided logic.

Reply:
We need to look at some better logic then as it got very messy and it is hard to comprehend what exactly this means while looking at link constants.

What if we have complete endpoints with placeholders. Like `xx/%s/yy`. The placeholder could be a `%s`. It would still be possible to have complete paths and also matching could be done (via a smarter function to match placeholders).

One could also even do something like `fmt.Sprintf(OpenApiEndpointVdcComputePolicies, id1, id2)` to fill in gaps.

Reply:
Issue here is MAP. In map we match endpoint and version. But some endpoints have dynamic part. For now I make it work if we agree it's ok.
```
endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcs + vdc.AdminVdc.ID + types.OpenApiEndpointAssignedComputePolicies
	minimumApiVersion, err := vdc.client.checkOpenApiEndpointCompatibility(types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcs + types.OpenApiEndpointAssignedComputePolicies)
```

Reply:
Up to you but I would think that ID placeholder could be reflected in the map. Map lookup can have a function and can even be inside a struct with our own methods to easen such lookups/matching. Now it is not clear which endpoint is bound to specific version. Looking at constants it is not immediately clear what endpoint should be:

```go
const (
OpenApiEndpointVdcComputePolicies      = "vdcComputePolicies/"
OpenApiEndpointAssignedComputePolicies = "/computePolicies"
OpenApiEndpointVdcs                    = "vdcs/"

OpenApiEndpointVdcAssignedComputePolicies = "vdcs/%s/computePolicies" // <- reflects complete endpoint
)
```

Looking at path in map now it looks like: 
```go
var endpointMinApiVersions = map[string]string{
	...
	types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcs + types.OpenApiEndpointAssignedComputePolicies: "33.0",
}
```
If we had a placeholder that could become:
```go
var endpointMinApiVersions = map[string]string{
	...
	types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcAssignedComputePolicies: "33.0",
}
```
And one could fill in the gaps like:

```go
OpenApiBuildEndpoint(types.OpenApiPathVersion1_0_0, fmt.Sprintf(types.OpenApiEndpointVdcAssignedComputePolicies, vdcId))
````

Reply:
I am not sure, but let's try. Changed to it.

Reply:
This looks a lot better for me. Thanks